src/coreclr/jit/instr.h

```diff
@@ -330,6 +339,31 @@ enum insBarrier : unsigned
 };
 #endif
 
+#if defined(TARGET_XARCH)
+// Represents tupletype attribute of instruction.
+// This is used in determining factor N while calculating compressed displacement in EVEX encoding
+// Reference: Section 2.6.5 in Intel 64 and ia-32 architectures software developer's manual volume 2.
+enum insTupleType : uint32_t
+{
+    INS_TT_NONE             = 0x00000,
+    INS_TT_FULL             = 0x00001,
+    INS_TT_HALF             = 0x00002,
+    INS_TT_IS_BROADCAST     = 0x00003,
+    INS_TT_FULL_MEM         = 0x00010,
+    INS_TT_TUPLE1_SCALAR    = 0x00020,
+    INS_TT_TUPLE1_FIXED     = 0x00040,
+    INS_TT_TUPLE2           = 0x00080,
+    INS_TT_TUPLE4           = 0x00100,
+    INS_TT_TUPLE8           = 0x00200,
+    INS_TT_HALF_MEM         = 0x00400,
+    INS_TT_QUARTER_MEM      = 0x00800,
+    INS_TT_EIGHTH_MEM       = 0x01000,
+    INS_TT_MEM128           = 0x02000,
+    INS_TT_MOVDDUP          = 0x04000,
+    INS_TT_IS_NON_BROADCAST = 0x7FFFC
```

Comment:
A few minor nits to consider for later:

- The high zero isn't needed, especially since there aren't any cases which set it. Remove it?
- Thus, the "7" in INS_TT_IS_NON_BROADCAST is not needed
- I was confused at first by INS_TT_IS_BROADCAST being 3 and not 4. Maybe rename INS_TT_IS_BROADCAST to INS_TT_BROADCAST_MASK and INS_TT_IS_NON_BROADCAST to INS_TT_NON_BROADCAST_MASK to make it clear they are bit masks? (I notice they currently aren't used)